### DISCUSSION TITLE
signature and checksum mismatch

### DISCUSSION THREAD

**Discussion by lachlanbelcher** [nwchemgit/nwchem#727] [General]
Hi gang,
I'm working on some new features and I'm coming across some MA errors.  I'm a bit lost.  What's the best procedure for tracking down these memory problems?

```
MA_verify_allocator_stuff: starting scan ...
stack block 'int_1e_sifs:sifbuf', handle 32, address 0x37050aa8:
        current right signature 2162688 != proper right signature 1431655765
stack block ' Ð<93>\<82><89>z?^N', handle 13, address 0x37050b28:
        current checksum 18386279124451654504 != stored checksum 4605689410543662570
```

Thanks!

**Comment by jeffhammond** [comment]
need input and output file.  i've seen this when i allocate a huge amount of memory and use all of it.

**Comment by lachlanbelcher** [comment]
Thanks.  Based on the debug comments in the output file (khe_nwaoints.out), it looks like it is happening inside int_so_sifs.F; however, the error messages seem to be from memory blocks initialized in int_1e_sifs.F (which calls int_so_sifs.F).  I will attach both to see if they are any help.
[khe_nwaoints.out.txt](https://github.com/nwchemgit/nwchem/files/10512187/khe_nwaoints.out.txt)
[int_1e_sifs.F.txt](https://github.com/nwchemgit/nwchem/files/10512188/int_1e_sifs.F.txt)
[int_so_sifs.F.txt](https://github.com/nwchemgit/nwchem/files/10512189/int_so_sifs.F.txt)
[NW_sonadc_tests.oe.txt](https://github.com/nwchemgit/nwchem/files/10512197/NW_sonadc_tests.oe.txt)

**Reply by lachlanbelcher** [reply]
See approx line 145 in int_so_sifs.f and line 137 in int_1e_sifs.F

**Comment by edoapra** [comment]
The best way to debug this kind of failures is to introduce the call to `ma_verify_allocator_stuff()` in as many places as possible until to hit the right place where the bug is.
Here is a code snippet

```
...
#include "mafdecls.fh"
...
      if (.not.ma_verify_allocator_stuff()) stop ' place 1'
....
     if (.not.ma_verify_allocator_stuff()) stop '  place 2'
     do i=1,1000
        if (.not.ma_verify_allocator_stuff()) stop ' place 3'
        a(i)= ....
        if (.not.ma_verify_allocator_stuff()) stop ' place 4'
     enddo
      if (.not.ma_verify_allocator_stuff()) stop ' place 5'
```

**Reply by lachlanbelcher** [reply]
Thank you!  I will give this a try

**Comment by edoapra** [comment]
All this activities should happen with the use of `gdb`, if possible!
